src/source/PeerConnection/Rtcp.c

```diff
@@ -199,34 +199,28 @@ STATUS onRtcpPacket(PKvsPeerConnection pKvsPeerConnection, PBYTE pBuff, UINT32 b
 STATUS onRtcpRembPacket(PRtcpPacket pRtcpPacket, PKvsPeerConnection pKvsPeerConnection)
 {
     STATUS retStatus = STATUS_SUCCESS;
+    STATUS tmpRetStatus = STATUS_SUCCESS;
     UINT32 ssrcList[MAX_UINT8] = {0};
     DOUBLE maximumBitRate = 0;
     UINT8 ssrcListLen;
     UINT32 i;
-    PDoubleListNode pCurNode = NULL;
     PKvsRtpTransceiver pTransceiver = NULL;
-    UINT64 item;
 
     CHK(pKvsPeerConnection != NULL && pRtcpPacket != NULL, STATUS_NULL_ARG);
 
     CHK_STATUS(rembValueGet(pRtcpPacket->payload, pRtcpPacket->payloadLength, &maximumBitRate, (PUINT32) &ssrcList, &ssrcListLen));
 
     for (i = 0; i < ssrcListLen; i++) {
-        CHK_STATUS(doubleListGetHeadNode(pKvsPeerConnection->pTransceivers, &pCurNode));
-        while (pCurNode != NULL && pTransceiver == NULL) {
-            CHK_STATUS(doubleListGetNodeData(pCurNode, &item));
-            CHK(item != 0, STATUS_INTERNAL_ERROR);
-
-            pTransceiver = (PKvsRtpTransceiver) item;
-            if (pTransceiver->sender.ssrc != ssrcList[i] && pTransceiver->sender.rtxSsrc != ssrcList[i]) {
-                pTransceiver = NULL;
+        pTransceiver = NULL;
+        tmpRetStatus = findTransceiverBySsrc(pKvsPeerConnection, &pTransceiver, ssrcList[i]);
+        if (STATUS_FAILED(tmpRetStatus)) {
+            DLOGE("Received REMB for non existing ssrcs: ssrc %lu", ssrcList[i]);
```

Comment:
Not sure why we are returning an error here? Perhaps, just DLOGW printout is enough and skipping the transceiver? I am not sure if this adds much value to return an error, though I can see it both ways. The core SDK won't process the error and it will be dumped to the ground anyway.

If you decide to go with the DLOGW then you can simply skip over the transceiver and continue.

Reply:
makes sense